app/views/advanced_messenger/_html_head.html.erb

```diff
@@ -32,7 +32,34 @@
                     $("#all-unread-notifications-indicator").removeClass('ajax-loading'); 
                 },
                 success: function(result) {
-                    renderGlobalUnreadNotificationsIndicator(result.count)
+                    if (result.notifications && result.notifications.length > 0) {
+                        // Get the stored notifications from sessionStorage (those already notified)
+                        let storedNotifications = JSON.parse(sessionStorage.getItem('PWA_NOTIFICATIONS') || '[]');
+
+                        // Filter out notifications that have already been read
+                        const unreadNotificationIds = result.notifications.map(n => n.notificationId);
+                        const filteredStoredNotifications = storedNotifications.filter(storedNotification =>
+                            unreadNotificationIds.includes(storedNotification.notificationId)
+                        );
+
+                        // Update sessionStorage to only store unread notifications
+                        sessionStorage.setItem('PWA_NOTIFICATIONS', JSON.stringify(filteredStoredNotifications));
+
+                        // Filter out notifications that have already been notified
+                        var newNotifications = result.notifications.filter(function (notification) {
+                            return !storedNotifications.some(function (storedNotification) {
```

Comment:
Aici cred ca puteai folosi pe `filteredStoredNotifications` in loc de stored notifications daca tot l-am calculat mai sus

Reply:
Da, am modificat.